### PR TITLE
Add eslint infrastructure

### PR DESCRIPTION
**Pull request by loay**

### CODE REVIEW

**Comment by loay** [comment]
Hi @bajtos 
please take a look. Thanks

File: package.json [hunk 1]
```diff
@@ -25,7 +27,9 @@
     "passport-oauth2": "^1.1.2",
     "passport-twitter": "^1.0.2",
     "serve-favicon": "^2.0.1",
-    "loopback-component-explorer": "^2.1.0"
+    "loopback-component-explorer": "^2.1.0",
+    "eslint": "^2.5.3",
+    "eslint-config-loopback": "^1.0.0"
```
Comment (bajtos) [inline package.json]: Please use the latest version of our linting rules, I think it's `^4.0.0`? Just run `npm install --save-dev eslint-config-loopback@latest`.

You will need to revert some of the fixes after that, I'll wait with further review until you make that updates.

Also could you please send a PR updating https://github.com/strongloop-internal/loopback-knowledge-base/blob/4b0aacb99501e3d6d4b31a52685e1d14ab268c68/eslint-guide.md to make it clear that people should always use the latest version of our eslint config when converting a new repo?

Comment (loay) [inline package.json]: sure, I will work on both items.

**Comment by loay** [comment]
@bajtos PTAL

File: package.json [hunk 2]
```diff
@@ -25,11 +27,15 @@
     "passport-oauth2": "^1.1.2",
     "passport-twitter": "^1.0.2",
     "serve-favicon": "^2.0.1",
-    "loopback-component-explorer": "^2.1.0"
+    "loopback-component-explorer": "^2.1.0",
+    "eslint": "^2.5.3"
```
Comment (bajtos) [inline package.json]: Why is `eslint` a regular dependency? Please move it to `devDependencies` instead.

File: package.json [hunk 3]
```diff
@@ -25,11 +27,15 @@
...
   },
   "repository": {
     "type": "git",
     "url": "git://github.com/strongloop-community/loopback-example-passport.git"
   },
-  "license": "MIT"
+  "license": "MIT",
+  "devDependencies": {
```
Comment (bajtos) [inline package.json]: Please put `devDependencies` higher in the file, just after `dependencies`.

**Comment by bajtos** [comment]
Almost there 🎯 , two more comments to address.

**Comment by loay** [comment]
@bajtos done. PTAL

**Comment by bajtos** [comment]
LGTM, please squash the commits before landing.

**Pull request merged**